File: DataStruct/1.两数之和.cpp
#include <iostream>
#include <vector>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        for (int i = 0; i < nums.max_size(); i++) {
            for (int j = i; j < nums.max_size(); j++) {
                if (nums[i] + nums[j] == target) {
                    return { i,j };
                }
            }
        }
    }
};

File: DataStruct/LeetCode/1.两数之和.go
package LeetCode

func twoSum(nums []int, target int) []int {
	numMap := make(map[int]int, len(nums))
	var ans [2]int
	for i, v := range nums {
		if idx, ok := numMap[target-v]; ok {
			ans[0] = i
			ans[1] = idx
			return ans[:]
		}
		numMap[v] = i
	}
	return ans[:]
}
